docs/docs/resources/pipeline-components/kafka-connector.yaml

```
@@ -3,21 +3,37 @@
   # Pipeline prefix that will prefix every component name. If you wish to not
   # have any prefix you can specify an empty string.
   prefix: ${pipeline_name}-
-  # Topic(s) from which the component will read input
   from: # Must not be null
     topics: # read from topic
       ${pipeline_name}-input-topic:
-        type: input # required
-        # role: topic-role # only used if type is `extra` or `extra-pattern`
+        type: input # Implied when role is NOT specified
+      ${pipeline_name}-extra-topic:
+        role: topic-role # Implies `type` to be extra
+      ${pipeline_name}-input-pattern-topic:
+        type: pattern # Implied to be an input pattern if `role` is undefined
+      ${pipeline_name}-extra-pattern-topic:
+        type: pattern # Implied to be an extra pattern if `role` undefined
```

same as in other places
```suggestion
        type: pattern # Implied to be an extra pattern if `role` is defined
```

---

Ah, I missed that one. BTW [`sections/_from.yaml`](https://github.com/bakdata/kpops/blob/refactor/input-output-types/docs/docs/resources/pipeline-components/sections/from_.yaml#L10) is the only place where a manual change is needed. Such a relief!